Doc/library/exceptions.rst

```diff
@@ -965,9 +965,10 @@ their subgroups based on the types of the contained exceptions.
          def derive(self, excs):
             return Errors(excs, self.exit_code)
 
-   Like :exc:`ExceptionGroup`, any subclass of :exc:`BaseExceptionGroup` which
-   is also a subclass of :exc:`Exception` can only wrap instances of
-   :exc:`Exception`.
+   A subclass of :exc:`BaseExceptionGroup` that extends :exc:`Exception`
+   cannot wrap :exc:`BaseException`\ s, as is the case with :exc:`ExceptionGroup`.
+   However, there is no further restriction: a :exc:`BaseExceptionGroup` subclass
+   which extends :exc:`KeyError` could wrap any :exc:`Exception`.
```

Comment:
I don't think this comment is easy to understand without the justification. How about this?

```suggestion
   Like :exc:`ExceptionGroup`, user-defined subclasses of :exc:`BaseExceptionGroup`
   that extend :exc:`Exception` can only wrap :exc:`Exception` subclasses.
   This is to prevent exceptions that represent fatal errors from being wrapped in an
   exception group which can be caught by ``except Exception``. See also :ref:`exceptions`. 
```